dll/win32/iernonce/registry.cpp

```diff
@@ -37,21 +38,43 @@ BOOL RunOnceExEntry::Delete(
 
 BOOL RunOnceExEntry::Exec() const
 {
-    // CreateProcessW requires lpCommandLine is non-const, so copy one.
-    CStringW ValueCopy = m_Value;
-    STARTUPINFOW si = { sizeof(si) };
-    PROCESS_INFORMATION pi = { 0 };
-    BOOL bSuccess = CreateProcessW(NULL, ValueCopy.GetBuffer(), NULL, NULL, FALSE, 0, NULL, NULL, &si, &pi);
-    ValueCopy.ReleaseBuffer();
+    CStringW CommandLine;
+    if (wcsncmp(m_Value, L"||", 2) == 0)
+    {
+        // remove the prefix;
+        CommandLine = (LPCWSTR)m_Value + 2;
+    }
+    else
+    {
+        CommandLine = m_Value;
+    }
+
+    // FIXME: SHEvaluateSystemCommandTemplate is not implemented
+    //        using PathGetArgsW, PathRemoveArgsW as a workaround.
```

Comment:
iernonce of windows 2000 and <= SP1 windows xp or 2003 wouldn't use this API (since it's late SP2 and Vista+)

Reply:
Note also that ShellExecuteEx can do environment substitution (`SEE_MASK_DOENVSUBST` - Expand any environment variables specified in the string given by the lpDirectory or lpFile member.), and also has a `SEE_MASK_NOASYNC` flag.

Reply:
`SEE_MASK_DOENVSUBST` seems does not expands environment variables in `lpParameters` as documented, I guess we want it to be expanded as well, 😄 and we've already expanded it when reading from the registry.

Reply:
SEE_MASK_NOASYNC seems to be used to wait for some background process & DDE operations. I'm not sure if it's better to also specify it, but we still need to wait on `hProcess` anyway